src/BlazorWebFormsComponents/BaseWebFormsComponent.cs

```diff
@@ -226,21 +226,27 @@ public ValueTask DisposeAsync()
 
 		public BaseWebFormsComponent ()
 		{
+			// Get Access to the ComponentBase fields we need to wrap every component in a CascadingValue
+			var _renderFragmentField = typeof(ComponentBase).GetField("_renderFragment", BindingFlags.NonPublic | BindingFlags.Instance);
+			var _hasPendingQueuedRenderField = typeof(ComponentBase).GetField("_hasPendingQueuedRender", BindingFlags.NonPublic | BindingFlags.Instance);
+			var _hasNeverRenderedField = typeof(ComponentBase).GetField("_hasNeverRendered", BindingFlags.NonPublic | BindingFlags.Instance);
+
+			// Create a RenderFragment to go into the CascadingValue
 			RenderFragment _standardRender = builder2 => {
-				_hasPendingQueuedRender = false;
-				_hasNeverRendered = false;
+				_hasPendingQueuedRenderField.SetValue(this,false);
+				_hasNeverRenderedField.SetValue(this, false);
 				BuildRenderTree(builder2);
-				// Include this to see it working
-				//builder2.AddContent(9999,"All your bases are loaded");
 			};
-			_renderFragment = builder =>
+
+			// Override the default RenderFragment with our Special Sauce version
+			_renderFragmentField.SetValue(this, (RenderFragment)(builder =>
 				{
 					builder.OpenComponent(1, typeof(CascadingValue<BaseWebFormsComponent>));
 					builder.AddAttribute(2, "Name", "ParentComponent");

```

Comment:
To make things more refactor friendly, I suggest changing the Parent property to this:

```csharp
		[CascadingParameter(Name=nameof(Parent))]
		public virtual BaseWebFormsComponent Parent { get; set; }
```

Then the follow could be:

```suggestion
					builder.AddAttribute(2, "Name", nameof(Parent));
					builder.AddAttribute(5, "Fixed", true); // This should be possible here as well
```

Reply:
Yes, good catch!

Reply:
Actually, you could even remove the other magic strings also:

```csharp
builder.AddAttribute(2, nameof(CascadingValue<BaseWebFormsComponent>.Name), nameof(Parent));
```